nft-marketplace/marketplace/io/src/lib.rs

```diff
@@ -0,0 +1,158 @@
+#![no_std]
+use codec::{Decode, Encode};
+use gstd::{prelude::*, ActorId};
+use primitive_types::U256;
+use scale_info::TypeInfo;
+use sp_core::H256 as spH256;
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub struct InitMarket {
+    pub owner_id: ActorId,
+    pub treasury_id: ActorId,
+    pub treasury_fee: u128,
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo, Clone)]
+pub struct Offer {
+    pub hash: spH256,
+    pub id: ActorId,
+    pub ft_contract_id: Option<ActorId>,
+    pub price: u128,
+}
+
+#[derive(Debug, Default, Encode, Decode, TypeInfo, Clone)]
+pub struct Auction {
+    pub bid_period: u64,
+    pub started_at: u64,
+    pub ended_at: u64,
+    pub current_price: u128,
+    pub bids: Option<Vec<Bid>>,
```

Comment:
Does this really need to be Option? Empty vectory doesn't mean the same as `Option::None` in this case?